source/s2n/s2n_tls_channel_handler.c

```diff
@@ -573,6 +588,12 @@ static int s_s2n_handler_process_read_message(
         } else {
             aws_mem_release(outgoing_read_message->allocator, outgoing_read_message);
         }
+
+        if (blocked == S2N_NOT_BLOCKED) {
+            /* read is complete so shutdown with success */
+            aws_channel_shutdown(slot->channel, AWS_OP_SUCCESS);
+            return AWS_OP_SUCCESS;
+        }
```

Comment:
this is not correct
I'm pretty sure S2N_NOT_BLOCKED means we've processed all available data, so we can exit the loop, but we might get more data from the socket in the future

it doesn't mean the connection is over

Reply:
Yeah, I have fixed it. I confused return value `0` and `read > 0 + NOT_BLOCKED`.